### ISSUE TITLE
Dropdown: Change event is triggered on keyboard navigation

### ISSUE THREAD

**Issue by MatviiStelmakh** [primefaces/primeng#12819]
### Describe the feature you would like to see added

The change event should be triggered on mouse click or keyboard Enter event, not the keyboard navigation.
The current behavior is not intuitive and very inconvenient if you have an action on this event like changing pagination size which leads to a page reload.

Similar requests were closed without implementation.
https://github.com/primefaces/primeng/issues/5335
https://github.com/primefaces/primeng/issues/9864
https://github.com/primefaces/primeng/issues/11703

Still, for v.15.2, I see that a change event is emitted on keyboard navigation. 
https://stackblitz.com/edit/angular-wzpqmh

### Describe the solution you'd like

Instead of introducing a breaking change to the current logic, it will be useful to have additional event like `onSelect($event)` where the event is the selected option.
